industry_benchmarks/utils/fix_networks.py

```diff
@@ -184,8 +184,24 @@ def _check_and_deduplicate_transforms(
 
     # Only adds transformations if mappings are present twice, meaning both
     # solvent and complex phases are present in the transform_list
+    # Check if both solvent and complex legs finished
     mappings = [t.mapping for t in transform_list]
-    transform_list = [e for inx, e in enumerate(transform_list) if mappings.count(mappings[inx]) == 2]
+    for inx, e in enumerate(transform_list):
+        if mappings.count(mappings[inx]) != 2:
+            errmsg = (
+                "Only results from one leg (either complex or solvent) found "
+                f"for {t_name}. This indicates a partially completed set of "
```

Comment:
We need to find the name of the transformation somehow as `t_name` is from the last loop.

Reply:
This one is a bit tricky, as in this case the name would not be in `t_list` as there are no results. I could either just assume that the name is the same as in the other leg, just switching out `solvent` to `complex` and vise versa. 
If we wanted to get the name from the files themselves, we would have to pass in as an argument the original alchemical network. What do you think?

Reply:
What about just reusing the names from above, we could make a list of names and append each name of the transform while adding the transform
```python
transform_list.append(t_list[0])
transform_names.append(t_name)
```
then we can loop over them
```python
for inx, t_name in enumerate(transform_names):
...
```

Reply:
But since in this error case there wouldn't be a single result for that leg, the name wouldn't be in this list, right? So I think it would only be in the full alchemical network, or am I misunderstanding this?

Reply:
Good point I think my solution results in the name of the wrong file being shown like `solvent_liganda_ligandb` when this is the leg that worked. What about extracting the name of the ligand from the edge like so
```python
# build the name of the edge from the ligands
name = f"{e.stateA.components['ligand'].name}_{e.stateB.components['ligand'].name}"
```